test/xpu_api/random/interface_tests/engines_methods.pass.cpp

```diff
@@ -270,7 +291,15 @@ test
                         Engine engine1;
                         engine1.seed(SEED);
                         engine0.discard(offset);
+                        if (engine0 == engine1)
+                        {
+                            dpstd_acc[offset] = 0;
```

Comment:
I suggest to make sure that dpstd_res is filled with 0, change get_access<sycl::access::mode::write> to get_access<sycl::access::mode::read_write> and make dpstd_acc[offset] += 1; instead if there is an error (otherwise this output will be overwritten on line 312)